### PR TITLE
Adding error announcements to user form

### PR DESCRIPTION
**Pull request by dfmcphee**
This fixes https://github.com/CovidShield/portal/issues/7.

It uses the suggestion in that issue to connect the error messages to the inputs using aria

### CODE REVIEW

File: app/views/users/_form.html.erb [hunk 1]
```diff
@@ -12,9 +12,13 @@
...
             <%= form.label :username, t('username'), class: 'text-field__label' %>
-            <%= form.text_field :username, class: 'text-field__input' %>
+            <%= form.text_field :username,
+              class: 'text-field__input',
+              'aria-describedby': (user.errors[:username].any? ? 'username-error' : nil),
+              'aria-invalid': (user.errors[:username].any? ? true : nil)
+            %>
             <% if user.errors[:username].any? %>
-              <div class="text-field__error-message">
+              <div id="usrname-error" class="text-field__error-message">
```
Comment (svinkle) [inline app/views/users/_form.html.erb:21]: Spelling: `username-error`.

**Comment by svinkle** [comment]
I forgot to mention, I'd suggest changing each `.text-field__error-message` containers from `div` to the more semantically correct [`small` element](https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-small-element).

**Comment by dfmcphee** [comment]
Good catch. Updated with that. 👍

**Comment by svinkle** [comment]
@dfmcphee Would it be possible to send focus to the first `input` in an error state? This would be helpful in conveying the current state.

And now that I think about it, there's no explicit "success" message other than loading a new page. I guess there was nothing in the design for a success state visual?

**Comment by dfmcphee** [comment]
> @dfmcphee Would it be possible to send focus to the first `input` in an error state? This would be helpful in conveying the current state.

For sure. Added that now. Will follow up about a success message in another PR.

**Pull request merged**